org.csstudio.display.builder.model/src/org/csstudio/display/builder/model/widgets/SlideButtonWidget.java

```diff
@@ -0,0 +1,181 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.csstudio.display.builder.model.widgets;
+
+import static org.csstudio.display.builder.model.properties.CommonWidgetProperties.newBooleanPropertyDescriptor;
+import static org.csstudio.display.builder.model.properties.CommonWidgetProperties.newStringPropertyDescriptor;
+import static org.csstudio.display.builder.model.properties.CommonWidgetProperties.propBit;
+import static org.csstudio.display.builder.model.properties.CommonWidgetProperties.propConfirmDialogOptions;
+import static org.csstudio.display.builder.model.properties.CommonWidgetProperties.propConfirmMessage;
+import static org.csstudio.display.builder.model.properties.CommonWidgetProperties.propEnabled;
+import static org.csstudio.display.builder.model.properties.CommonWidgetProperties.propFont;
+import static org.csstudio.display.builder.model.properties.CommonWidgetProperties.propForegroundColor;
+import static org.csstudio.display.builder.model.properties.CommonWidgetProperties.propOffColor;
+import static org.csstudio.display.builder.model.properties.CommonWidgetProperties.propOnColor;
+import static org.csstudio.display.builder.model.properties.CommonWidgetProperties.propPassword;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.csstudio.display.builder.model.Messages;
+import org.csstudio.display.builder.model.Widget;
+import org.csstudio.display.builder.model.WidgetCategory;
+import org.csstudio.display.builder.model.WidgetDescriptor;
+import org.csstudio.display.builder.model.WidgetProperty;
+import org.csstudio.display.builder.model.WidgetPropertyCategory;
+import org.csstudio.display.builder.model.WidgetPropertyDescriptor;
+import org.csstudio.display.builder.model.persist.NamedWidgetColors;
+import org.csstudio.display.builder.model.persist.NamedWidgetFonts;
+import org.csstudio.display.builder.model.persist.WidgetColorService;
+import org.csstudio.display.builder.model.persist.WidgetFontService;
+import org.csstudio.display.builder.model.properties.ConfirmDialog;
+import org.csstudio.display.builder.model.properties.WidgetColor;
+import org.csstudio.display.builder.model.properties.WidgetFont;
+
+/**
+ * Widget that provides button for making a binary change.
+ *
+ * @author claudiorosati, European Spallation Source ERIC
+ * @version 1.0.0 21 Aug 2018
+ */
+public class SlideButtonWidget extends WritablePVWidget {
+
+    /**
+     * Widget descriptor
+     */
+    public static final WidgetDescriptor WIDGET_DESCRIPTOR =
+        new WidgetDescriptor("slide_button", WidgetCategory.CONTROL,
+            "Slide Button",
+            "platform:/plugin/org.csstudio.display.builder.model/icons/slide_button.png",
+            "Slide button that can toggle one bit of a PV value between 1 and 0",
+            Arrays.asList( "org.csstudio.opibuilder.widgets.checkbox"))
```

Comment:
Do you intent for this to replace legacy checkbox widgets, which are already handled by https://github.com/kasemir/org.csstudio.display.builder/blob/master/org.csstudio.display.builder.model/src/org/csstudio/display/builder/model/widgets/CheckBoxWidget.java#L46 ?

Or do you mean for this to handle legacy the legacy "org.csstudio.opibuilder.widgets.BoolSwitch"?

Reply:
Ohoops,
I just started from the CheckBox and this was a leftover.
I think this is a more modern alternative to CheckBox, but was not my intention to replace it.
I'll remove line 57 and the corresponding entry in the plugin file.

Reply:
About the legacy, it can have an image, and this is already handled by the Boolean Button